sources/packages/backend/apps/api/src/route-controllers/institution/_tests_/e2e/institution.aest.controller.searchInstitutions.e2e-spec.ts

```diff
@@ -0,0 +1,147 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeInstitution,
+} from "@sims/test-utils";
+import { SearchInstitutionAPIOutDTO } from "../../models/institution.dto";
+
+describe("InstitutionAESTController(e2e)-searchInstitutions", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  const FAKE_INSTITUTION_NAME = "Search E2E Test institution name";
+  const FAKE_INSTITUTION_LEGAL_NAME = "Search E2E Test institution legal name";
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should return institution(s) when searched by legal name.", async () => {
+    // Arrange
+    const institution = createFakeInstitution();
+    institution.operatingName = FAKE_INSTITUTION_NAME;
+    institution.legalOperatingName = FAKE_INSTITUTION_LEGAL_NAME;
+    await db.institution.save(institution);
+    const legalNameSearchText = "Search E2E Test institution legal";
+    const mailingAddress = institution.institutionAddress.mailingAddress;
+    const expectedSearchResult: SearchInstitutionAPIOutDTO[] = [
+      {
+        id: institution.id,
+        legalName: institution.legalOperatingName,
+        operatingName: institution.operatingName,
+        address: {
+          addressLine1: mailingAddress.addressLine1,
+          addressLine2: mailingAddress.addressLine2,
+          city: mailingAddress.city,
+          provinceState: mailingAddress.provinceState,
+          country: mailingAddress.country,
+          postalCode: mailingAddress.postalCode,
+        },
+      },
+    ];
+
+    const endpoint = `/aest/institution/search?legalName=${legalNameSearchText}`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const searchResult = response.body as SearchInstitutionAPIOutDTO[];
+        expect(searchResult).toEqual(
+          expect.arrayContaining(expectedSearchResult),
+        );
+      });
+  });
+
+  it("Should return institutions when searched by legal name and operating name.", async () => {
+    // Arrange
+    const institution = createFakeInstitution();
+    institution.operatingName = FAKE_INSTITUTION_NAME;
+    institution.legalOperatingName = FAKE_INSTITUTION_LEGAL_NAME;
+    await db.institution.save(institution);
+    const searchText = "Search E2E Test institution";
+    const mailingAddress = institution.institutionAddress.mailingAddress;
+    const expectedSearchResult: SearchInstitutionAPIOutDTO[] = [
```

Comment:
just wondering, do we need to specify the type here and in 101, and for other TCs

Reply:
I am not sure about if we need to. I do not see that as blocker in other TCs. But IMO it feels better to have it typed for many reasons. One of them is, if the DTO gets modified, it becomes more evident here on what needs to be changed.

Reply:
Yup, just wanted to know, Not sure if we are following in all e2e, I thought we were typing only when it's needed. but I am good with it. I am approving the PR

Reply:
IMO I would prefer them to not be typed (using API types) and, in general, we consume as little as possible from the API types. That would be the same idea around trying to avoid even some API constants and having the E2E tests use more "raw" asserts.
If the DTOs change in a way that a test will break we will know quickly 😉